cmd/manager/main.go

```diff
@@ -106,16 +107,30 @@ func main() {
 		os.Exit(1)
 	}
 
+	if err := monitoringv1.AddToScheme(mgr.GetScheme()); err != nil {
+		log.Error(err, "error registering prometheus monitoring objects")
+		os.Exit(1)
+	}
+
 	// Setup all Controllers
 	if err := controller.AddToManager(mgr); err != nil {
 		log.Error(err, "")
 		os.Exit(1)
 	}
 
-	// Create Service object to expose the metrics port.
-	_, err = metrics.ExposeMetricsPort(ctx, metricsPort)
+	s, _ := operatormetrics.GenerateService(8080, "metrics")
```

Comment:
I think it would be valuable to do something with the error returned from `GenerateService` instead of using `_` to discard it. Specifically, this is where an error is being thrown in another operator I'm working on, and by default mine used `_`, which hid the error.

Reply:
You can see mine here, if you're interested. It just lets the user know why the operator is crashing. https://github.com/openshift/configure-alertmanager-operator/pull/18/files